modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+        ComplianceFeature complianceFeature = new ComplianceFeature();
+        complianceFeature.setFeature(activeFeature);
+        complianceFeature.setFeatureCode(activeFeature.getFeatureCode());
+        complianceFeature.setCompliance(deviceFeature.isCompliance());
+        return complianceFeature;
+    }
+
+    /**
+     * Update the completed/Error status of the operation which have the URI of the operation code in the syncml payload.
+     *
+     * @param syncmlDocument SyncmlDocument object generated from the the syncml engine.
+     * @throws DeviceManagementException
+     * @throws NotificationManagementException
+     * @throws OperationManagementException
+     */
+    public void UpdateUriOperations(SyncmlDocument syncmlDocument) throws DeviceManagementException,
+            NotificationManagementException, OperationManagementException, WindowsOperationException {
+        DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(
+                syncmlDocument.getHeader().getSource().getLocURI());
+        List<Status> statuses = syncmlDocument.getBody().getStatus();
+        OperationUtils operationUtils = new OperationUtils();
+
+        for (Status status : statuses) {
+
+            if (status.getCommand().equals(Constants.EXECUTE)) {
+                if (status.getTargetReference() == null) {
+                    operationUtils.updateDeviceOperations(status, syncmlDocument, deviceIdentifier);
+                } else {
+                    if (status.getTargetReference().equals(OperationCode.Command.DEVICE_LOCK)) {
+                        operationUtils.lockOperationUpdate(status, syncmlDocument, deviceIdentifier);
+                    }
+                    if (status.getTargetReference().equals(OperationCode.Command.DEVICE_RING)) {
+                        operationUtils.ring(status, syncmlDocument, deviceIdentifier);
+                    }
+                    if (status.getTargetReference().equals(OperationCode.Command.WIPE_DATA)) {
+                        operationUtils.dataWipe(status, syncmlDocument, deviceIdentifier);
+                    }
+                }
+            }
+            if (status.getCommand().equals(Constants.SEQUENCE)) {
+                if (status.getData().equals(Constants.SyncMLResponseCodes.ACCEPTED)) {
+
+                    pendingDataOperations = SyncmlUtils.getDeviceManagementService()
+                            .getOperationsByDeviceAndStatus(deviceIdentifier, Operation.Status.PENDING);
+                    for (Operation operation : pendingDataOperations) {
+                        if (operation.getCode().equals(PluginConstants.OperationCodes.POLICY_BUNDLE) &&
+                                operation.getId() == status.getCommandReference()) {
+                            operation.setStatus(Operation.Status.COMPLETED);
+                        }
+                        if (operation.getCode().equals(PluginConstants.OperationCodes.MONITOR) &&
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.